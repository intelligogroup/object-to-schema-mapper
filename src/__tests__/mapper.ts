
import data = require('./data.json');
import { mapObjectToSchema } from '../objectToSchemaTransformer';

describe('mapper tests', () => {

    test('mapper maps object to schema correctly', () => {

        expect(mapObjectToSchema(data)).toMatchSnapshot('inputDataMapper 1');
    });

})