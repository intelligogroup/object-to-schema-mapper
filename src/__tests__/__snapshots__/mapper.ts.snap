// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapper tests mapper maps object to schema correctly: inputDataMapper 1 1`] = `
Object {
  "__v": "{\\"type\\":\\"number\\"}",
  "_id": "{\\"type\\":\\"number\\"}",
  "_type": "{\\"type\\":\\"string\\"}",
  "createdAt": "{\\"type\\":\\"string\\"}",
  "data": Object {
    "addresses": Array [
      Object {
        "addressHash": "{\\"type\\":\\"string\\"}",
        "addressOrder": "{\\"type\\":\\"number\\"}",
        "addressQualityCodes": Array [
          Object {},
        ],
        "city": "{\\"type\\":\\"string\\"}",
        "county": "{\\"type\\":\\"string\\"}",
        "firstReportedDate": "{\\"type\\":\\"string\\"}",
        "fullAddress": "{\\"type\\":\\"string\\"}",
        "houseNumber": "{\\"type\\":\\"string\\"}",
        "isDeliverable": "{\\"type\\":\\"boolean\\"}",
        "isMergedAddress": "{\\"type\\":\\"boolean\\"}",
        "lastReportedDate": "{\\"type\\":\\"string\\"}",
        "latitude": "{\\"type\\":\\"string\\"}",
        "longitude": "{\\"type\\":\\"string\\"}",
        "neighborSummaryRecords": Array [
          Object {
            "city": "{\\"type\\":\\"string\\"}",
            "emailAddress": "unknown type",
            "fullAddress": "{\\"type\\":\\"string\\"}",
            "fullName": "{\\"type\\":\\"string\\"}",
            "houseNumber": "{\\"type\\":\\"string\\"}",
            "phone": "{\\"type\\":\\"string\\"}",
            "state": "{\\"type\\":\\"string\\"}",
            "streetName": "{\\"type\\":\\"string\\"}",
            "streetPostDirection": "{\\"type\\":\\"string\\"}",
            "streetPreDirection": "{\\"type\\":\\"string\\"}",
            "streetType": "{\\"type\\":\\"string\\"}",
            "tahoeId": "{\\"type\\":\\"string\\"}",
            "unit": "{\\"type\\":\\"string\\"}",
            "unitType": "unknown type",
            "zip": "{\\"type\\":\\"string\\"}",
            "zip4": "{\\"type\\":\\"string\\"}",
          },
        ],
        "neighbors": Array [
          Object {},
        ],
        "phoneNumbers": Array [
          Object {},
        ],
        "publicFirstSeenDate": "{\\"type\\":\\"string\\"}",
        "state": "{\\"type\\":\\"string\\"}",
        "streetName": "{\\"type\\":\\"string\\"}",
        "streetPostDirection": "{\\"type\\":\\"string\\"}",
        "streetPreDirection": "{\\"type\\":\\"string\\"}",
        "streetType": "{\\"type\\":\\"string\\"}",
        "totalFirstSeenDate": "{\\"type\\":\\"string\\"}",
        "unit": "{\\"type\\":\\"string\\"}",
        "unitType": "unknown type",
        "zip": "{\\"type\\":\\"string\\"}",
        "zip4": "{\\"type\\":\\"string\\"}",
      },
    ],
    "age": "{\\"type\\":\\"number\\"}",
    "akas": Array [
      Object {
        "firstName": "{\\"type\\":\\"string\\"}",
        "lastName": "{\\"type\\":\\"string\\"}",
        "middleName": "{\\"type\\":\\"string\\"}",
        "prefix": "{\\"type\\":\\"string\\"}",
        "publicFirstSeenDate": "{\\"type\\":\\"string\\"}",
        "suffix": "{\\"type\\":\\"string\\"}",
        "totalFirstSeenDate": "{\\"type\\":\\"string\\"}",
      },
    ],
    "associatesSummary": Array [
      Object {
        "dob": "{\\"type\\":\\"string\\"}",
        "firstName": "{\\"type\\":\\"string\\"}",
        "lastName": "{\\"type\\":\\"string\\"}",
        "middleName": "{\\"type\\":\\"string\\"}",
        "prefix": "{\\"type\\":\\"string\\"}",
        "score": "{\\"type\\":\\"number\\"}",
        "suffix": "{\\"type\\":\\"string\\"}",
        "tahoeId": "{\\"type\\":\\"string\\"}",
      },
    ],
    "bankruptcySearchIds": Array [
      Object {},
    ],
    "businessSummaryRecords": Array [
      Object {},
    ],
    "datesOfBirth": Array [
      Object {
        "age": "{\\"type\\":\\"number\\"}",
        "dob": "{\\"type\\":\\"string\\"}",
      },
    ],
    "datesOfDeath": Array [
      Object {},
    ],
    "deathRecords": Object {
      "isDeceased": "{\\"type\\":\\"boolean\\"}",
    },
    "debtV2Records": Array [
      Object {
        "addresses": Array [
          Object {
            "addressHash": "{\\"type\\":\\"number\\"}",
            "city": "{\\"type\\":\\"string\\"}",
            "county": "{\\"type\\":\\"string\\"}",
            "fullAddress": "{\\"type\\":\\"string\\"}",
            "houseNumber": "{\\"type\\":\\"string\\"}",
            "latitude": "{\\"type\\":\\"string\\"}",
            "longitude": "{\\"type\\":\\"string\\"}",
            "rawVerificationCodes": "{\\"type\\":\\"string\\"}",
            "state": "{\\"type\\":\\"string\\"}",
            "streetDirection": "unknown type",
            "streetName": "{\\"type\\":\\"string\\"}",
            "streetPostDirection": "unknown type",
            "streetType": "unknown type",
            "type": "{\\"type\\":\\"string\\"}",
            "unitNumber": "unknown type",
            "unitType": "unknown type",
            "verificationCodes": Array [
              "{\\"type\\":\\"STRING\\"}",
            ],
            "zip4": "{\\"type\\":\\"string\\"}",
            "zipCode": "{\\"type\\":\\"string\\"}",
          },
        ],
        "caseDetails": Array [
          Object {
            "actionType": "unknown type",
            "actionTypeDescription": "unknown type",
            "amountOrLiability": "unknown type",
            "assetValue": "unknown type",
            "assets": "unknown type",
            "assetsAvailable": "unknown type",
            "assetsAvailableFlag": "unknown type",
            "associateCode": "unknown type",
            "associationCode": "unknown type",
            "attorney": "unknown type",
            "attorneyZipCode": "unknown type",
            "bankruptcyDismissalFlag": "unknown type",
            "bankruptcyType": "unknown type",
            "book": "{\\"type\\":\\"string\\"}",
            "caseNumber": "{\\"type\\":\\"string\\"}",
            "caseNumber2": "unknown type",
            "chapter": "unknown type",
            "claimDate": "unknown type",
            "closedDate": "unknown type",
            "convertedDate": "unknown type",
            "courtState": "unknown type",
            "dischargeDate": "unknown type",
            "dismissalDate": "unknown type",
            "entity": "unknown type",
            "fileState": "{\\"type\\":\\"string\\"}",
            "filingDate": "{\\"type\\":\\"string\\"}",
            "filingType": "{\\"type\\":\\"string\\"}",
            "filingTypeDescription": "{\\"type\\":\\"string\\"}",
            "generationCode": "unknown type",
            "judgesInitials": "unknown type",
            "lawFirm": "unknown type",
            "liability": "unknown type",
            "liabilityAmount": "{\\"type\\":\\"string\\"}",
            "noticeDate": "unknown type",
            "objectionDate": "unknown type",
            "originalBook": "unknown type",
            "originalCase": "unknown type",
            "originalCaseNumber": "unknown type",
            "originalDepartment": "unknown type",
            "originalPage": "unknown type",
            "otherCaseNumber": "unknown type",
            "page": "{\\"type\\":\\"string\\"}",
            "plaintiffLawFirm": "unknown type",
            "primaryIndustryBusiness": "unknown type",
            "releaseDate": "unknown type",
            "reopenedDate": "unknown type",
            "reportDate": "unknown type",
            "schedule341Date": "unknown type",
            "schedule341Time": "unknown type",
            "secondaryIndustryBusiness": "unknown type",
            "sourceType": "{\\"type\\":\\"string\\"}",
            "transferredDate": "unknown type",
            "unlawfulDetainer": "unknown type",
            "withdrawnDate": "unknown type",
          },
        ],
        "courts": Array [
          Object {},
        ],
        "debtType": "{\\"type\\":\\"string\\"}",
        "filingDate": "{\\"type\\":\\"string\\"}",
        "names": Array [
          Object {
            "address": Object {
              "addressHash": "{\\"type\\":\\"number\\"}",
              "city": "{\\"type\\":\\"string\\"}",
              "county": "{\\"type\\":\\"string\\"}",
              "fullAddress": "{\\"type\\":\\"string\\"}",
              "houseNumber": "{\\"type\\":\\"string\\"}",
              "latitude": "{\\"type\\":\\"string\\"}",
              "longitude": "{\\"type\\":\\"string\\"}",
              "rawVerificationCodes": "{\\"type\\":\\"string\\"}",
              "state": "{\\"type\\":\\"string\\"}",
              "streetDirection": "unknown type",
              "streetName": "{\\"type\\":\\"string\\"}",
              "streetPostDirection": "unknown type",
              "streetType": "unknown type",
              "type": "{\\"type\\":\\"string\\"}",
              "unitNumber": "unknown type",
              "unitType": "unknown type",
              "verificationCodes": Array [
                "{\\"type\\":\\"STRING\\"}",
              ],
              "zip4": "{\\"type\\":\\"string\\"}",
              "zipCode": "{\\"type\\":\\"string\\"}",
            },
            "addressHash": "{\\"type\\":\\"number\\"}",
            "businessName": "unknown type",
            "ein": "unknown type",
            "email": "unknown type",
            "firstName": "{\\"type\\":\\"string\\"}",
            "fullName": "{\\"type\\":\\"string\\"}",
            "isBusiness": "{\\"type\\":\\"boolean\\"}",
            "lastName": "{\\"type\\":\\"string\\"}",
            "middleName": "{\\"type\\":\\"string\\"}",
            "nameHash": "{\\"type\\":\\"number\\"}",
            "phone": "unknown type",
            "prefix": "unknown type",
            "ssn": "{\\"type\\":\\"string\\"}",
            "suffix": "unknown type",
            "tahoeId": "{\\"type\\":\\"string\\"}",
            "type": "{\\"type\\":\\"string\\"}",
          },
        ],
        "poseidonId": "{\\"type\\":\\"number\\"}",
        "reportDate": "{\\"type\\":\\"string\\"}",
      },
    ],
    "debtV2SearchIds": Array [
      "{\\"type\\":\\"STRING\\"}",
    ],
    "divorceRecords": Array [
      Object {
        "caseBookId": "{\\"type\\":\\"string\\"}",
        "certificatNo": "{\\"type\\":\\"string\\"}",
        "county": "{\\"type\\":\\"string\\"}",
        "decree": "{\\"type\\":\\"number\\"}",
        "divorceDate": "{\\"type\\":\\"string\\"}",
        "fileId": "{\\"type\\":\\"string\\"}",
        "marriageDate": "{\\"type\\":\\"string\\"}",
        "otherSpouseAge": "{\\"type\\":\\"number\\"}",
        "otherSpouseFirstName": "{\\"type\\":\\"string\\"}",
        "otherSpouseFullName": "{\\"type\\":\\"string\\"}",
        "otherSpouseLastName": "{\\"type\\":\\"string\\"}",
        "otherSpouseMiddleName": "{\\"type\\":\\"string\\"}",
        "otherSpouseNameSuffix": "{\\"type\\":\\"string\\"}",
        "poseidonId": "{\\"type\\":\\"string\\"}",
        "recordSubType": "{\\"type\\":\\"string\\"}",
        "recordType": "{\\"type\\":\\"string\\"}",
        "spouseAge": "{\\"type\\":\\"number\\"}",
        "spouseFirstName": "{\\"type\\":\\"string\\"}",
        "spouseGender": "{\\"type\\":\\"string\\"}",
        "spouseKey": "{\\"type\\":\\"string\\"}",
        "spouseLastName": "{\\"type\\":\\"string\\"}",
        "spouseMaidenName": "{\\"type\\":\\"string\\"}",
        "spouseMiddleName": "{\\"type\\":\\"string\\"}",
        "spouseNameSuffix": "{\\"type\\":\\"string\\"}",
        "ssn": "{\\"type\\":\\"string\\"}",
        "state": "{\\"type\\":\\"string\\"}",
        "tahoeId": "{\\"type\\":\\"string\\"}",
      },
    ],
    "divorceSearchIds": Array [
      "{\\"type\\":\\"STRING\\"}",
    ],
    "dob": "{\\"type\\":\\"string\\"}",
    "domainRecords": Array [
      Object {},
    ],
    "emailAddresses": Array [
      Object {},
    ],
    "fullName": "{\\"type\\":\\"string\\"}",
    "isOptedOut": "{\\"type\\":\\"boolean\\"}",
    "isPublic": "{\\"type\\":\\"boolean\\"}",
    "judgmentSearchIds": Array [
      Object {},
    ],
    "lienSearchIds": Array [
      "{\\"type\\":\\"STRING\\"}",
    ],
    "locations": Array [
      Object {
        "city": "{\\"type\\":\\"string\\"}",
        "state": "{\\"type\\":\\"string\\"}",
      },
    ],
    "marriageRecords": Array [
      Object {
        "caseBookId": "{\\"type\\":\\"string\\"}",
        "certificatNo": "{\\"type\\":\\"string\\"}",
        "county": "{\\"type\\":\\"string\\"}",
        "decree": "{\\"type\\":\\"number\\"}",
        "divorceDate": "unknown type",
        "fileId": "{\\"type\\":\\"string\\"}",
        "marriageDate": "{\\"type\\":\\"string\\"}",
        "otherSpouseAge": "{\\"type\\":\\"number\\"}",
        "otherSpouseFirstName": "{\\"type\\":\\"string\\"}",
        "otherSpouseFullName": "{\\"type\\":\\"string\\"}",
        "otherSpouseLastName": "{\\"type\\":\\"string\\"}",
        "otherSpouseMiddleName": "{\\"type\\":\\"string\\"}",
        "otherSpouseNameSuffix": "{\\"type\\":\\"string\\"}",
        "poseidonId": "{\\"type\\":\\"string\\"}",
        "recordSubType": "{\\"type\\":\\"string\\"}",
        "recordType": "{\\"type\\":\\"string\\"}",
        "spouseAge": "{\\"type\\":\\"number\\"}",
        "spouseFirstName": "{\\"type\\":\\"string\\"}",
        "spouseGender": "{\\"type\\":\\"string\\"}",
        "spouseKey": "{\\"type\\":\\"string\\"}",
        "spouseLastName": "{\\"type\\":\\"string\\"}",
        "spouseMaidenName": "{\\"type\\":\\"string\\"}",
        "spouseMiddleName": "{\\"type\\":\\"string\\"}",
        "spouseNameSuffix": "{\\"type\\":\\"string\\"}",
        "ssn": "{\\"type\\":\\"string\\"}",
        "state": "{\\"type\\":\\"string\\"}",
        "tahoeId": "{\\"type\\":\\"string\\"}",
      },
    ],
    "marriageSearchIds": Array [
      "{\\"type\\":\\"STRING\\"}",
    ],
    "mergedNames": Array [
      Object {
        "firstName": "{\\"type\\":\\"string\\"}",
        "lastName": "{\\"type\\":\\"string\\"}",
        "middleName": "{\\"type\\":\\"string\\"}",
        "prefix": "{\\"type\\":\\"string\\"}",
        "publicFirstSeenDate": "unknown type",
        "suffix": "{\\"type\\":\\"string\\"}",
        "totalFirstSeenDate": "unknown type",
      },
    ],
    "name": Object {
      "firstName": "{\\"type\\":\\"string\\"}",
      "lastName": "{\\"type\\":\\"string\\"}",
      "middleName": "{\\"type\\":\\"string\\"}",
      "prefix": "{\\"type\\":\\"string\\"}",
      "publicFirstSeenDate": "{\\"type\\":\\"string\\"}",
      "suffix": "{\\"type\\":\\"string\\"}",
      "totalFirstSeenDate": "{\\"type\\":\\"string\\"}",
    },
    "phoneNumbers": Array [
      Object {
        "company": "{\\"type\\":\\"string\\"}",
        "firstReportedDate": "{\\"type\\":\\"string\\"}",
        "isConnected": "{\\"type\\":\\"boolean\\"}",
        "lastReportedDate": "{\\"type\\":\\"string\\"}",
        "latitude": "{\\"type\\":\\"string\\"}",
        "location": "{\\"type\\":\\"string\\"}",
        "longitude": "{\\"type\\":\\"string\\"}",
        "phoneNumber": "{\\"type\\":\\"string\\"}",
        "phoneOrder": "{\\"type\\":\\"number\\"}",
        "phoneType": "{\\"type\\":\\"string\\"}",
        "publicFirstSeenDate": "unknown type",
        "totalFirstSeenDate": "{\\"type\\":\\"string\\"}",
      },
    ],
    "propertySummaryRecords": Array [
      Object {
        "apn": "{\\"type\\":\\"string\\"}",
        "coOwnerName": "{\\"type\\":\\"string\\"}",
        "coOwnerTahoeId": "{\\"type\\":\\"string\\"}",
        "coSellerName": "{\\"type\\":\\"string\\"}",
        "coSellerTahoeId": "unknown type",
        "fullAddress": "{\\"type\\":\\"string\\"}",
        "hasRecorderRecord": "{\\"type\\":\\"boolean\\"}",
        "isCurrentlyOwned": "{\\"type\\":\\"boolean\\"}",
        "ownerName": "{\\"type\\":\\"string\\"}",
        "ownerTahoeId": "{\\"type\\":\\"string\\"}",
        "propertyCity": "{\\"type\\":\\"string\\"}",
        "propertyId": "{\\"type\\":\\"string\\"}",
        "propertyLocation": "{\\"type\\":\\"string\\"}",
        "propertyState": "{\\"type\\":\\"string\\"}",
        "propertyZip": "{\\"type\\":\\"string\\"}",
        "purchaseAmount": "{\\"type\\":\\"string\\"}",
        "purchaseDate": "{\\"type\\":\\"string\\"}",
        "saleAmount": "{\\"type\\":\\"string\\"}",
        "saleDate": "{\\"type\\":\\"string\\"}",
        "sellerName": "{\\"type\\":\\"string\\"}",
        "sellerTahoeId": "unknown type",
        "siteAddressHash": "{\\"type\\":\\"string\\"}",
      },
    ],
    "relativesSummary": Array [
      Object {
        "dob": "{\\"type\\":\\"string\\"}",
        "firstName": "{\\"type\\":\\"string\\"}",
        "lastName": "{\\"type\\":\\"string\\"}",
        "middleName": "{\\"type\\":\\"string\\"}",
        "prefix": "{\\"type\\":\\"string\\"}",
        "relativeLevel": "{\\"type\\":\\"string\\"}",
        "relativeType": "{\\"type\\":\\"string\\"}",
        "score": "{\\"type\\":\\"number\\"}",
        "sharedHouseholdIds": Array [
          Object {},
        ],
        "spouse": "{\\"type\\":\\"number\\"}",
        "suffix": "{\\"type\\":\\"string\\"}",
        "tahoeId": "{\\"type\\":\\"string\\"}",
      },
    ],
    "socialSecurityNumbers": Array [
      Object {
        "ssn": "{\\"type\\":\\"string\\"}",
      },
    ],
    "ssn": "{\\"type\\":\\"string\\"}",
    "ssnAlternateLocation": "{\\"type\\":\\"string\\"}",
    "ssnDates": "{\\"type\\":\\"string\\"}",
    "ssnLocation": "{\\"type\\":\\"string\\"}",
    "tahoeId": "{\\"type\\":\\"string\\"}",
  },
  "resultId": "{\\"type\\":\\"number\\"}",
  "runId": "{\\"type\\":\\"string\\"}",
  "scraperId": "{\\"type\\":\\"number\\"}",
  "searchId": "{\\"type\\":\\"number\\"}",
  "status": "{\\"type\\":\\"string\\"}",
  "updatedAt": "{\\"type\\":\\"string\\"}",
}
`;
